### PR TITLE
448 improved Github actions

### PR DESCRIPTION
**Pull request by rc10house**
Adds improved github actions for CI/CD #448. Specifically linting, formatting, and automatic license checking. Still will need to add type checking, automatic release notes, and cronjobs for triggering release candidates in the future.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -1,34 +1,64 @@
...
 
 name: Node.js CI
-
+run-name: ${{ github.actor }} running CI suite
 on:
   push:
     branches:
       - master
       - test
+      - 448-improve-GH-actions #TODO: remove this
```
Comment (EandrewJones) [inline .github/workflows/node.js.yml:30]: This needs to be removed, no?

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -1,34 +1,64 @@
...
           node-version: ${{ matrix.node-version }}
-      - run: npm ci
-      - run: npm run test
\ No newline at end of file
+      - run: npm ci # clean install
+      - run: npm run format
+      - run: npm run build
+      - run: npm run lint
+      # TODO: add type checking
+      # TODO: add skywalking-eyes for licensing
```
Comment (EandrewJones) [inline .github/workflows/node.js.yml:54]: Needs to be removed.

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -1,34 +1,64 @@
...
+      - 448-improve-GH-actions #TODO: remove this
   pull_request:
     branches:
       - master
       - test
-
 jobs:
-  build:
-
+  CI-tests:
```
Comment (EandrewJones) [inline .github/workflows/node.js.yml:36]: Can we:
1) treat linting+type checking and building+testing as two different jobs
2) Label steps with names

See [ECharts GH actions](https://github.com/apache/echarts/blob/master/.github/workflows/ci.yml), for example.

**Review by EandrewJones — CHANGES_REQUESTED** [review]

**Comment by rc10house** [comment]
Resolved requested changes

**Comment by rc10house** [comment]
Made requested changes

**Review by EandrewJones — APPROVED** [review]

**Pull request merged**